feat(datasource/galaxy-collection): support ansible automation hub (#25675)
renovatebot/renovate: lib/modules/datasource/galaxy-collection/readme.md

> By default, the `galaxy-collection` datasource checks for dependencies on `https://galaxy.ansible.com`.
> 
> But you can set your own registries.
> You can set a `source` in your `requirements.yaml` file, _or_ write a `packageRule` to set a new `registryURLs`.
> This allows you to use Renovate with a private automation hub.

Reviewer: ```suggestion
By default, the `galaxy-collection` datasource checks for dependencies on `https://galaxy.ansible.com`.
But you can override the default if you want.

Set your own registries by:

- setting a `source` in your `requirements.yaml` file, _or_
- writing a `packageRule` to set a new `registryURLs`

Then you can use Renovate with a private automation hub.
```

How about this? I think a list is easier to understand. I've also moved the sentences about the default and overriding into a single paragraph. Plus rewritten some sentences.

> By default the `ansible-galaxy` collection will lookup your dependencies on https://galaxy.ansible.com.

Reviewer: ```suggestion
By default, the `galaxy-collection` datasource checks for dependencies on `https://galaxy.ansible.com`.
```

I think the name of the datasource is `galaxy-collection`, so we should use that.

> However, you can override that behavior by either specifying a `source` in your `requirements.yaml` file or write a `packageRule` to provide other registryURLs.

Reviewer: ```suggestion
But you can set your own registries.
You can set a `source` in your `requirements.yaml` file, _or_ write a `packageRule` to set a new `registryURLs`.
```

> This allows you to use renovate with a private automation hub.

Reviewer: ```suggestion
This allows you to use Renovate with a private automation hub.
```

> **Usage example - requirements.yaml**
> 
> ```yaml

Reviewer: ```suggestion
```yaml title="Example config for requirements.yaml"
```

We should use codeblock titles here, as they look better in the published docs.

[Material for MkDocs manual, codeblock titles](https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#adding-a-title)

---
collections:
  - name: community.general
    version: 3.0.0
    source: https://hub.mydomain.com/api/galaxy/content/community/
```

> **Usage example - renovate.json**
> 
> ```json

Reviewer: ```suggestion
```json title="Example config for renovate.json"
```
